src/init/CliRunner.ts

```diff
@@ -0,0 +1,57 @@
+import * as Path from 'path';
+import { ReadStream, WriteStream } from 'tty';
+import { Loader, LoaderProperties } from 'componentsjs';
+import yargs from 'yargs';
+import { RuntimeConfig } from './RuntimeConfig';
+import { Setup } from './Setup';
+
+/**
+ * Generic run function for starting the server from a given config
+ * @param args - Command line arguments.
+ * @param stdin - Standard input stream.
+ * @param stdout - Standard output stream.
+ * @param stderr - Standard error stream.
+ * @param properties - Components loader properties.
+ */
+export const runCustom = function(
+  args: string[],
+  stdin: ReadStream,
+  stdout: WriteStream,
+  stderr: WriteStream,
+  properties: LoaderProperties,
+): void {
+  const { argv } = yargs
+    .usage('node ./bin/server.js [args]')
+    .options({
+      port: { type: 'number', alias: 'p' },
+      config: { type: 'string', alias: 'c' },
+    })
+    .help();
+
+  new Promise<RuntimeConfig>(async(resolve): Promise<void> => {
```

Comment:
We need `runCustom` to be non-async, because we want to call it from `bin/` without having to add a then/catch there.

Reply:
But it's fake safety because it's completing non-synchronously anyway.

Reply:
It's safe though, since we handle `then` and `catch` directly here.

But in any case, I can remove the explicit `Promise` constructor as you've mentioned, and just call the `async` function directly.